### ISSUE TITLE
BigDecimal support

### ISSUE THREAD

**Issue by dinee432** [google/gson#812]
How do I specify to use bigdecimal to handle numbers in gson? Below is the sample code and sample json string that I am trying to convert to a map.

```
`Type type = new TypeToken<Map<String, Object>>() { }.getType();
  Map<String, Object> data = new Gson().fromJson(strInput, type);`
```

Sample Json string : 

[{
    "Type":"Test",
    "Amount":100231212111.00,  
}]

The result I am getting is a map with double value for Amount. Please help to resolve this issue

**Comment by lyubomyr-shaydariv** [comment]
* The JSON document structure you provided does not match the type you declared in the type token. It should be `List<Map<String, Object>>`.
* Unfortunately, as of Gson 2.8.5 you cannot tell Gson to convert `Object` to a specific `Number` implementation.

**Comment by eamonnmcmanus** [comment]
Fixed by #1290.
